docs/index.rst

```diff
@@ -8,36 +8,58 @@ HPy: a better API for Python
 
 HPy provides a new API for extending Python in C.
 
+There are several advantages to write your C extension in HPy:
+
+  - **Speed**: it runs much faster on PyPy, GraalPy, and at native speed on CPython
+
+  - **Deployment**: it is possible to compile a single binary which runs unmodified on all
+    supported Python implementations and versions -- think "stable ABI" on steroids
+
+  - **Simplicity**: it is simpler and more manageable than the ``Python.h`` API, both for
+    the users and the Pythons implementing it
+
+  - **Debugging**: it provides an improved debugging experience. Debug mode can be turned
+    on at runtime without the need to recompile the extension or the Python running it.
+    HPy design is more suitable for automated checks.
+
 The official `Python/C API <https://docs.python.org/3/c-api/index.html>`_,
 also informally known as ``#include <Python.h>``, is
 specific to the current implementation of CPython: it exposes a lot of
-internal details which makes it hard:
+internal details which makes it hard to:
 
-  - to implement it for other Python implementations (e.g. PyPy, GraalPython,
-    Jython, IronPython, etc.)
+  - implement it for other Python implementations (e.g. PyPy, GraalPy,
+    Jython, ...)
 
-  - to experiment with new things inside CPython itself: e.g. using a GC
-    instead of refcounting, or to remove the GIL.
+  - experiment with new approaches inside CPython itself, for example:
 
-There are several advantages to write your C extension in HPy:
+    - use a tracing garbage collection instead of reference counting
+    - remove the global interpreter lock (GIL) to take full advantage of multicore architectures
```

Comment:
Is this true? Does using HPy make it any easier to be sure code does not have threading problems?

Reply:
Yes, good point. This bullet point was here before, I just shuffled it and added "... to take full advantage of multicore architectures".

> HPy make it any easier to be sure code does not have threading problems

Depends on the point of view. My line of thinking is that it would make it easier to do the internal changes in CPython that are needed to remove the GIL. Still extensions would have to make sure that they operate correctly without GIL.  

I think it still falls into the category of "what is wrong with current C API", although better HPy would not solve the problem as a whole, only a part of it. Explaining all this in detail would make this bullet point too long. I like that "removing GIL" is something that many would clearly understand brings benefits and there is bit of a hype around it now :-) I think this simplification could be fine for such a pitch as the beginning of the documentation?